Mathlib/RingTheory/AlgebraicIndependent/TranscendenceBasis.lean

```diff
@@ -125,3 +233,204 @@ theorem IsTranscendenceBasis.isAlgebraic_field {F E : Type*} {x : ι → E}
   haveI : IsScalarTower (adjoin F S) (IntermediateField.adjoin F S) E :=
     IsScalarTower.of_algebraMap_eq (congrFun rfl)
   exact Algebra.IsAlgebraic.extendScalars (R := adjoin F S) (Subalgebra.inclusion_injective _)
+
+namespace AlgebraicIndependent
+
+variable [NoZeroDivisors A] (inj : Injective (algebraMap R A))
+
+private def indepMatroid : IndepMatroid A where
+  E := univ
+  Indep s := AlgebraicIndependent R ((↑) : s → A)
+  indep_empty := (algebraicIndependent_empty_iff ..).mpr inj
+  indep_subset _ _ := (·.mono)
+  indep_aug I B I_ind h B_base := by
+    contrapose! h
+    rw [← isTranscendenceBasis_iff_maximal] at B_base ⊢
+    cases subsingleton_or_nontrivial R
+    · rw [isTranscendenceBasis_iff_of_subsingleton] at B_base ⊢
+      contrapose! h
+      have ⟨b, hb⟩ := B_base
+      exact ⟨b, ⟨hb, fun hbI ↦ h ⟨b, hbI⟩⟩, algebraicIndependent_of_subsingleton⟩
+    rw [I_ind.isTranscendenceBasis_iff_isAlgebraic]
+    replace B_base := B_base.isAlgebraic
+    rw [Subtype.range_val] at B_base ⊢
+    set RI := adjoin R I
+    set RB := adjoin R B
+    let RIB := adjoin RI B
+    let _ : Algebra RB RIB := (Subalgebra.inclusion
+      (T := RIB.restrictScalars R) <| adjoin_le <| by apply subset_adjoin).toAlgebra
+    have : IsScalarTower RB RIB A := .of_algebraMap_eq fun ⟨a, _⟩ ↦ show a = _ from rfl
+    have : Algebra.IsAlgebraic RIB A := .extendScalars (R := RB) (Subalgebra.inclusion_injective _)
+    have : Algebra.IsAlgebraic RI RIB := by
+      have : Injective (algebraMap RI A) := Subtype.val_injective
+      have := (isDomain_iff_noZeroDivisors_and_nontrivial RI).mpr ⟨this.noZeroDivisors _
+        (map_zero _) (map_mul _), (Subtype.range_val ▸ I_ind.aevalEquiv).symm.nontrivial⟩
+      rw [← Subalgebra.isAlgebraic_iff, isAlgebraic_adjoin_iff]
+      intro x hB
+      by_cases hI : x ∈ I
+      · exact isAlgebraic_algebraMap (⟨x, subset_adjoin hI⟩ : RI)
+      contrapose! h
+      exact ⟨x, ⟨hB, hI⟩, (insert_iff hI).mpr ⟨I_ind, h⟩⟩
+    exact .trans' (R := RI) (S := RIB) Subtype.val_injective
+  indep_maximal X _ I ind hIX := exists_maximal_algebraicIndependent I X hIX ind
+  subset_ground _ _ := subset_univ _
+
+/-- If `R` is a commutative ring and `A` is a commutative `R`-algebra with injective algebra map
+and no zero-divisors, then the `R`-algebraic independent subsets of `A` form a matroid. -/
+def matroid : Matroid A := (indepMatroid inj).matroid
```

Comment:
`def matroid : Matroid A := (indepMatroid inj).matroid.copyBase univ 
  (fun s ↦ IsTranscendenceBasis R ((↑) : s → A)) 
  (fun B ↦ by rw [Matroid.base_iff_maximal_indep, isTranscendenceBasis_iff_maximal]; rfl)`

for defeq with `M.Base`. 

(I haven't tested this)